scripts/doc_crosslinks/crosslink_redirect.html

```diff
@@ -39,31 +42,68 @@
+                    }
+
+                    if (eventStreamInfo['output'] === true) {
+                        params = params + "," + pkgBase + typeName + "ResponseHandler";
+                    }
+                }
+
+                return "(" + params + ")"
+            }
+
+            function isAsyncOnlyMethod(uid, typeName) {
+                return uidToAsyncOnlyOperations[uid] != null && uidToAsyncOnlyOperations[uid][typeName] != null;
+            }
+
+            function asyncClientName(uid) {
+                return uidToClientPrefix[uid] + "AsyncClient";
+            }
+
+            function syncClientName(uid) {
+                return uidToClientPrefix[uid] + "Client";
+            }
+
             var typeName = getQueryStringParameter("type", window.location.href);
             var operation = getQueryStringParameter("operation", window.location.href);
             var uid = getQueryStringParameter("uid", window.location.href);
-            var clientObjectName =  uidToClientName[uid];
+            var clientObjectName =  syncClientName(uid);
             var serviceName =  uidToServiceName[uid];
             var redirectSuffixLink = "index.html"
 
             if(clientObjectName != null && serviceName != null) {
                 redirectServiceSuffixLink = serviceFolderPath + serviceName + "/model/" + typeName + ".html" ;
                 if(checkResourceExists( baseUri.replace(/crosslink_redirect.*/gi, redirectServiceSuffixLink) )) {
                     redirectSuffixLink =  redirectServiceSuffixLink;
-                }
-                else
-                {
-                    redirectServiceSuffixLink =  serviceFolderPath + serviceName + "/" + clientObjectName + ".html" ;
+                } else {
+                    redirectServiceSuffixLink =  serviceFolderPath + serviceName + "/";
                     if(typeName != null) {
-                        methodName = camelize(typeName)
-                        requestObjectName = methodName.charAt(0).toUpperCase() + methodName.slice(1) + "Request"
+                        if (isAsyncOnlyMethod(uid, typeName)) {
+                            clientObjectName = asyncClientName(uid);
+                        }
+                        redirectServiceSuffixLink = redirectServiceSuffixLink + clientObjectName + ".html" ;
+                        methodName = camelize(typeName);
                         redirectServiceSuffixLink =
                             redirectServiceSuffixLink +
-                            "#" + methodName +
-                            "(software.amazon.awssdk.services." + serviceName + ".model." + requestObjectName + ")";
+                            "#" +
+                            methodName + requestSignature(uid, serviceName, typeName);
+                    } else {
+                        redirectServiceSuffixLink = redirectServiceSuffixLink + clientObjectName + ".html" ;
```

Comment:
The else part was not there before , why is it required now ?

Reply:
Originally, we always use the sync client name for `clientObjectName` regardless of operation. But not, we need to know if `typeName` is async only first, before we know which client to use